torch/csrc/jit/script/compiler.cpp

```diff
@@ -486,19 +517,26 @@ at::optional<std::vector<Value*>> tryMatchSchema(
     return matched_inputs;
 }
 
-
 static Value* tryEmitBuiltin(
-  const std::shared_ptr<Operator>& op,
-  std::stringstream& failure_messages,
-  const SourceRange& loc,
-  Graph& graph,
-  Symbol name,
-  at::ArrayRef<NamedValue> inputs,
-  at::ArrayRef<NamedValue> attributes) {
+    const std::shared_ptr<Operator>& op,
```

Comment:
tryEmitBuiltin is used by `graph->insert` when inserting nodes as well. So this modification will let _any_ pass put a node in by making up a world token. This is unsafe since passes that happen after the compiler are not going to call annotate effects. Furthermore, it _prevents_ those passes from actually passing a valid world token when creating an op. We should probably distinguish between calls within compiler.cpp which do implicitly insert the world token and calls from graph->insert which should not.

Reply:
Makes sense. I'll lift these changes out into a `tryEmitMutableBuiltin` function and just rely on callers to check whether the schema is mutable and dispatch to the right emit function if so.